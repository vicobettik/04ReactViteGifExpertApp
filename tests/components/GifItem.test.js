const { render, screen } = require("@testing-library/react")
const { GifItem } = require("../../src/components")

describe('Pruebas en GifItem', () => {

    const titulo = 'saitama';
    const url = 'https://url.com/';

    test('Debe hacer match con el snapcshot', () => {

        const { container } = render(<GifItem title={titulo} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con la url y el alt indicado', () => {

        render(<GifItem title={titulo} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(titulo);

    });

    test('Debe de mostrar el titulo en el componente', () => {
      
        render(<GifItem title={titulo} url={url} />);
        expect(screen.getByText(titulo)).toBeTruthy()

    })
    



})
