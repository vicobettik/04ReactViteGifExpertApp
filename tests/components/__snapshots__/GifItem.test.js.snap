// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapcshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://url.com/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
